### PR TITLE
adding command triggered signal and optimizing command registration

### PR DESCRIPTION
**Pull request by afshin**

### CODE REVIEW

File: src/commandpalette/plugin.ts [hunk 1]
```diff
@@ -58,6 +62,12 @@ function register(container: Container): void {
 
 class CommandPaletteManager implements ICommandPalette {
   /**
+   * A signal emitted when a shortcut is added to the manager.
```
Comment (blink1073) [inline src/commandpalette/plugin.ts]: Wrong docstring?

File: src/commandpalette/plugin.ts [hunk 2]
```diff
@@ -58,6 +62,12 @@ function register(container: Container): void {
...
+   */
+  get commandTriggered(): ISignal<CommandPaletteManager, void> {
```
Comment (blink1073) [inline src/commandpalette/plugin.ts]: Why not emit the `id`?

File: src/commandpalette/plugin.ts [hunk 3]
```diff
@@ -113,8 +132,24 @@ class CommandPaletteManager implements ICommandPalette {
     });
   }
 
+  private _commandHandler = (command: any) => {
+    this.commandTriggered.emit(command);
+    this._commandRegistry.execute(command.id, command.args);
+  };
```
Comment (sccolbert) [inline src/commandpalette/plugin.ts]: one space after this.

File: src/commandpalette/plugin.ts [hunk 4]
```diff
@@ -113,8 +132,24 @@ class CommandPaletteManager implements ICommandPalette {
     });
   }
 
+  private _commandHandler = (command: any) => {
```
Comment (sccolbert) [inline src/commandpalette/plugin.ts]: maybe a better argument name than `command`?

File: src/commandpalette/plugin.ts [hunk 5]
```diff
@@ -58,6 +62,12 @@ function register(container: Container): void {
 
 class CommandPaletteManager implements ICommandPalette {
   /**
+   * A signal emitted when a command is triggered by the palette.
+   */
+  get commandTriggered(): ISignal<CommandPaletteManager, { id: string, args: any }> {
+    return CommandPaletteManagerPrivate.commandTriggeredSignal.bind(this);
+  }
```
Comment (sccolbert) [inline src/commandpalette/plugin.ts]: one blank line after this. And getters should come after the constructor.

File: src/commandpalette/index.ts [hunk 6]
```diff
@@ -16,16 +16,30 @@ import {
...
+ *   register the following commands:
+ * 1. `'command-palette:focus-input'`, which takes no arguments and focuses the
+ *   search bar.
  */
 export
 interface ICommandPalette {
   /**
+   * A signal emitted when a command is triggered by the palette.
+   */
+  commandTriggered: ISignal<ICommandPalette, { id: string, args: any }>;
```
Comment (sccolbert) [inline src/commandpalette/index.ts]: these should be separated by 1 blank line. Signals first, then properties, then methods.

**Pull request merged**